File: online/LeetCodeCN/501~600/575.cpp
// 分糖果
// https://leetcode-cn.com/problems/distribute-candies/

// 简单逻辑

class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        int sum = candyType.size();
        int type = 0;
        sort(candyType.begin(), candyType.end());
        auto end = unique(candyType.begin(), candyType.end());
        return min(int(end - candyType.begin()), sum / 2);
    }
};

File: online/LeetCodeCN/501~600/575.go
// 分糖果
// https://leetcode-cn.com/problems/distribute-candies/

// 简单逻辑

func distributeCandies(candyType []int) int {
	sort.Slice(candyType, func(i, j int) bool {
		return candyType[i] < candyType[j]
	})

	ans := 0
	prev := int(1e7)
	for _, v := range candyType {
		if prev != v {
			ans++
			prev = v
		}
	}

	if ans < len(candyType)/2 {
		return ans
	}
	return int(len(candyType) / 2)
}
